scripts/scil_compute_metrics_stats_in_ROI.py

```diff
@@ -0,0 +1,112 @@
+standard deviation will also be weighted.
+"""
+
+import argparse
+import nibabel as nib
+import numpy as np
+import os
+import json
+
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_json_args,
+                             assert_inputs_exist)
+from scilpy.utils.filenames import split_name_with_nii
+from scilpy.utils.metrics_tools import get_roi_metrics_stats
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__,
+                                formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_mask',
+                   help='Mask volume file name, formatted in any nibabel ' +
+                        'supported format.\nCan be a binary mask or a ' +
+                        'weighted mask.')
+
+    g = p.add_mutually_exclusive_group(required=True)
+    g.add_argument('--metrics_dir',
+                   help='Metrics files directory. name of the directory ' +
+                        'containing the metrics files.')
+    g.add_argument('--metrics', dest='metrics_file_list', nargs='+',
+                   help='Metrics nifti file name. list of the names of the ' +
+                        'metrics file, in nifti format.')
+
+    p.add_argument('--bin', action='store_true',
+                   help='If set, will consider every value of the mask ' +
+                        'higher than 0 to be part of the mask, and set to 1 ' +
+                        '(equivalent weighting for every voxel).')
+
+    p.add_argument('--normalize_weights', action='store_true',
+                   help='If set, the weights will be normalized to the [0,1] '
+                        'range.')
+
+    add_overwrite_arg(p)
+    add_json_args(p)
+
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_mask, [os.listdir(args.metrics_dir),
+                                               args.metrics_file_list])
+
+    # Load mask and validate content depending on flags
+    mask_img = nib.load(args.in_mask)
+
+    if not issubclass(mask_img.get_data_dtype().type, np.floating) and \
+            not args.normalize_weights:
+        parser.error('The mask file must contain floating point numbers.')
+
+    # Can be a weighted image
+    mask_data = mask_img.get_fdata(dtype=np.float32)
+
+    if args.normalize_weights:
+        mask_data /= np.sum(mask_data)
+
+    if np.min(mask_data) < 0.0 or np.max(mask_data) > 1.0:
+        parser.error('Mask data should only contain values between 0 and 1. '
+                     'Try --normalize_weights.')
+
+    if args.bin:
+        mask_data[np.where(mask_data > 0.0)] = 1.0
+
+    # Load all metrics files.
+    if args.metrics_dir:
+        metrics_files = [nib.load(args.metrics_dir + f)
+                         for f in sorted(os.listdir(args.metrics_dir))]
+    elif args.metrics_file_list:
```

Comment:
Technically, just else is enough, but ok if it's for readability. But my pycharm doesn't like it in line 96, it thinks metrics_file might not have been set.
`else: # args.metrics_file_list is not None`

Reply:
I'm going to keep it.